llvm/lib/Target/TargetLoweringObjectFile.cpp

```diff
@@ -47,6 +49,16 @@ void TargetLoweringObjectFile::Initialize(MCContext &ctx,
   // Reset various EH DWARF encodings.
   PersonalityEncoding = LSDAEncoding = TTypeEncoding = dwarf::DW_EH_PE_absptr;
   CallSiteEncoding = dwarf::DW_EH_PE_uleb128;
+
+  // In the CHERI pure-capability ABI we write capabilities to for the landing
+  // pads so the section must be relocated.
+  // XXX: Would be nice if there was a ELF::SHF_RELRO/SHF_INITIALIZED_DATA
+  //   so I don't also have to modify lld.
+  if (isCheriPurecapABI(TM)) {
+    // TODO: Could we (ab)use SHF_OS_NONCONFORMING
+    auto *LSDASectionELF = static_cast<MCSectionELF *>(LSDASection);
+    LSDASectionELF->setFlags(LSDASectionELF->getFlags() | ELF::SHF_WRITE);
```

Comment:
Is this not being called for the failing test?

Reply:
Yeah, it's not, turns out TLOF is only for CodeGen and most of the lower-level tools don't (and can't due to lack of language-level options etc for a TargetMachine) construct one. I have pushed things into MCAsmInfo now which is probably where it belongs (there's at least a CodePointerSize field in there so doesn't seem like a violation).